addons/binding/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/internal/convention/homeassistant/ComponentBinarySensor.java

```diff
@@ -43,17 +43,17 @@
         protected String payload_off = "OFF";
     };
 
-    public ComponentBinarySensor(CFactory.ComponentConfiguration builder) {
-        super(builder, ChannelConfiguration.class);
+    public ComponentBinarySensor(CFactory.ComponentConfiguration componentConfiguration) {
+        super(componentConfiguration, ChannelConfiguration.class);
 
         if (channelConfiguration.force_update) {
             throw new UnsupportedOperationException("Component:Sensor does not support forced updates");
         }
 
-        channels.put(sensorChannelID,
-                new CChannel(this, sensorChannelID, new OnOffValue(channelConfiguration.payload_on, channelConfiguration.payload_off),
-                        channelConfiguration.state_topic, null, channelConfiguration.name, channelConfiguration.unit_of_measurement,
-                        builder.getUpdateListener()));
+        buildChannel(sensorChannelID, new OnOffValue(channelConfiguration.payload_on, channelConfiguration.payload_off),
+                channelConfiguration.name).listener(componentConfiguration.getUpdateListener())//
+                        .stateTopic(channelConfiguration.state_topic, channelConfiguration.value_template)//
+                        .unit(channelConfiguration.unit_of_measurement)//
```

Comment:
Why those //? Does it help to convince Eclipse to do some proper formatting ;-)

Reply:
Yes, only formatting.
It helps understanding to have logical groups here